Formula/zrok.rb

```diff
@@ -0,0 +1,31 @@
+require "language/node"
+
+class Zrok < Formula
+  desc "Geo-scale, next-generation sharing platform built on top of OpenZiti"
+  homepage "https://zrok.io"
+  url "https://github.com/openziti/zrok.git",
+    tag: "v0.3.5", revision: "917891e9b1fbe28e112ff7e399a92d60080c79e4"
+  license "Apache-2.0"
+  head "https://github.com/openziti/zrok.git", branch: "main"
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+
+  def install
+    cd buildpath/"ui" do
+      system "npm", "install", *Language::Node.local_npm_install_args
+      system "npm", "run", "build"
+    end
+    cd buildpath do
+      ldflags = ["-X github.com/openziti/zrok/build.Version=#{version}",
+                 "-X github.com/openziti/zrok/build.Hash=#{Utils.git_head}"]
+      system "go", "build", *std_go_args(ldflags: ldflags), "github.com/openziti/zrok/cmd/zrok"
+    end
+  end
+
+  test do
+    version_output = shell_output("#{bin}/zrok version")
+    assert_match(version.to_s, version_output)
+    assert_match(/[[a-f0-9]{40}]/, version_output)
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.